pkg/k8s/k8s_service.go

```diff
@@ -142,3 +144,92 @@ func prepareService(
 	}
 	return svc, nil
 }
+
+func (c *Client) WaitForService(ctx context.Context, name string) error {
+	for {
+		service, err := c.clientset.CoreV1().Services(c.namespace).Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
+			return ErrGettingService.WithParams(name).Wrap(err)
+		}
+
+		if service.Spec.Type == v1.ServiceTypeLoadBalancer {
```

Comment:
These concurrent if conditions make me think a switch statement would be more useful and also asks the question how to handle the case where it is not one of these 2?

```go
switch service.Spec.Type {
case v1.ServiceTypeLoadBalance:
case v1.ServiceTypeNodePort: 
default: // What to do here??
}
```

Reply:
fixed